File: DSA/searching/linear_search.py
"""
Checks if a given element is present in a list or not.
"""

def linear_search(list, element):
    """
    Returns the index of the element if present in the list, else returns -1.
    """
    for i in range(len(list)):
        if list[i] == element:
            return i
    return -1

if __name__ == "__main__":
    list = [1, 2, 3, 4, 5, 6, 7, 8, 9]
    element = 4
    print(linear_search(list, element))

File: DSA/searching/linear_search.cpp
#include <iostream>

using namespace std;

int linear_search(int array[], int length, int element){
    int result = -1;
    for (int i = 0, found = 0; i< length && found == 0; i++){
        if (array[i] == element){
            result = i;
            found = 1;
        }
    }
    return result;
}

int main(){
    int array[10] = {1,2,3,4,5,6,7,8,9,10};

    cout << linear_search(array, 10, 4) << endl;
    cout << linear_search(array, 10, 10) << endl;
    cout << linear_search(array, 10, 0) << endl;
    cout << linear_search(array, 10, 55) << endl;
    cout << linear_search(array, 10, 2) << endl;
}